example/src/main/java/com/stripe/example/view/CreditCardView.java

```diff
@@ -0,0 +1,560 @@
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.widget.Space;
+import android.text.Editable;
+import android.text.Layout;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.util.DisplayMetrics;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+
+import com.stripe.android.model.Card;
+import com.stripe.android.util.StripeTextUtils;
+import com.stripe.example.R;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Created by simonkenny on 05/12/2016.
+ */
+
+public class CreditCardView extends FrameLayout {
+
+    public static final int ERROR_NONE = 0;
+    public static final int ERROR_NUMBER = 1;
+    public static final int ERROR_EXPIRY_MONTH = 2;
+    public static final int ERROR_EXPIRY_YEAR = 3;
+    public static final int ERROR_CVC = 4;
+    public static final int ERROR_UNKNOWN = 5;
+
+    /**
+     * Callback interface, set using setCallback(Callback callback)
+     *
+     * Required to make this useful
+     */
+    public interface Callback {
+        void onValidated(Card card);
+        void onError(int errorCode);
+        void onClearError();
+    }
+
+    //Some static helper methods, would be best to put these in a Util class
+
+    private static int sScreenWidth = 0;
+
+    public static int getScreenWidth(Context context) {
+        if (sScreenWidth == 0) {
+            WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+            DisplayMetrics dm = new DisplayMetrics();
+            wm.getDefaultDisplay().getMetrics(dm);
+            if (context.getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
+                sScreenWidth = Math.max(dm.widthPixels, dm.heightPixels);
+            } else {
+                sScreenWidth = Math.min(dm.widthPixels, dm.heightPixels);
+            }
+        }
+        return sScreenWidth;
+    }
+
+    public static int getImageResForCardBrand(String brand){
+        if (brand.equals(Card.AMERICAN_EXPRESS)) {
+            return R.drawable.stp_card_amex;
+        }
+        if (brand.equals(Card.DINERS_CLUB)) {
+            return R.drawable.stp_card_diners;
+        }
+        if (brand.equals(Card.DISCOVER)) {
+            return R.drawable.stp_card_discover;
+        }
+        if (brand.equals(Card.JCB)) {
+            return R.drawable.stp_card_jcb;
+        }
+        if (brand.equals(Card.MASTERCARD)) {
+            return R.drawable.stp_card_mastercard;
+        }
+        if (brand.equals(Card.VISA)) {
+            return R.drawable.stp_card_visa;
+        }
+        return R.drawable.stp_card_placeholder_template;
+
+    }
+
+    CustomHorizontalScrollView mScrollView;
+    Space mSpaceInContainer;
```

Comment:
Yes, I had originally used ButterKnife but striped it out when adding to the example. Will make private.